From fb10787e922f16f8c73eac0e179ac9b4875a36ce Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@krgm.moe>
Date: Fri, 26 Jun 2026 22:15:24 +0800
Subject: [PATCH 3/6] AOSCOS: kcm_mouse: move check for isAnonymousInputDevice
 up to prevent broken layout

On X11, this property was never supported, having the visibility property
inside the FormEntry would leave the label for the Device, General, and
Scroll Speed combo boxes visible even when the options are not available.

Move the visible tag up a level to prevent this.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 kcms/mouse/ui/main.qml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kcms/mouse/ui/main.qml b/kcms/mouse/ui/main.qml
index 8898157e85..7f9789f21a 100644
--- a/kcms/mouse/ui/main.qml
+++ b/kcms/mouse/ui/main.qml
@@ -61,10 +61,10 @@ KCMUtils.SimpleKCM {
         Kirigami.FormGroup {
             // Device
             Kirigami.FormEntry {
+                visible: !root.backend.isAnonymousInputDevice
                 contentItem: QQC2.ComboBox {
                     id: deviceSelector
                     Kirigami.FormData.label: i18ndc("kcmmouse", "@title:listbox select device", "Device:")
-                    visible: !root.backend.isAnonymousInputDevice
                     enabled: count > 1
                     Layout.fillWidth: true
                     model: root.backend.inputDevices
@@ -84,11 +84,11 @@ KCMUtils.SimpleKCM {
 
             // General
             Kirigami.FormEntry {
+                visible: !root.backend.isAnonymousInputDevice
                 contentItem: QQC2.CheckBox {
                     id: deviceEnabled
                     Kirigami.FormData.label: i18ndc("kcmmouse", "@title:group", "General:")
                     text: i18ndc("kcmmouse", "@option:check enable this mouse device", "Enable device")
-                    visible: !root.backend.isAnonymousInputDevice
                     enabled: root.device?.supportsDisableEvents ?? false
                     checked: root.device && (!root.device.supportsDisableEvents || root.device.enabled)
 
@@ -246,12 +246,12 @@ KCMUtils.SimpleKCM {
 
             Kirigami.FormEntry {
                 // Scroll Speed aka scroll Factor
+                visible: !root.backend.isAnonymousInputDevice
                 contentItem: GridLayout {
                     Kirigami.FormData.label: i18ndc("kcmmouse", "@label:slider and @label:spinbox", "Scrolling speed:")
                     Kirigami.FormData.buddyFor: scrollFactor
                     Layout.fillWidth: true
 
-                    visible: !root.backend.isAnonymousInputDevice
                     columns: 3
 
                     QQC2.Slider {
-- 
2.52.0

